packages/docusaurus-theme-live-codeblock/src/theme/Playground/index.js

```diff
@@ -11,13 +11,55 @@ import clsx from 'clsx';
 
 import styles from './styles.module.css';
 
-function Playground({children, theme, transformCode, ...props}) {
+const typescriptExtensions = ['ts', 'tsx'];
+const transformJs = (code) => `${code};`;
+
+function Playground({children, theme, className, scope}) {
+  const [tsTranspileError, setTsTranspileError] = React.useState(null);
+  const transformTs = React.useCallback((code) => {
+    try {
+      // eslint-disable-next-line global-require
+      const {code: transformed} = require('@babel/standalone').transform(code, {
+        filename: 'transformedCode.ts',
+        presets: [
+          'react',
+          [
+            'typescript',
+            {
+              isTSX: true,
+              allExtensions: true,
+            },
+          ],
+        ],
+      });
+      setTsTranspileError(null);
+
+      return transformed;
+    } catch (e) {
+      setTsTranspileError(e.message);
+    }
+
+    return '() => null;';
```

Comment:
in case if something goes wrong in the transpilation, render nothing. Theoretically it should never reach here.